src/txkube/testing/test/test_matchers.py

```diff
@@ -89,6 +78,45 @@ def test_mismatch(self):
         )
 
 
+    def test_mismatch_py2(self):
+        if str(int) != "<type 'int'>":
```

Comment:
As a Python version check, this seems fragile and unnecessarily indirect.  Since a mismatch here means the test is skipped (instead of the test failing), it also seems like an easy problem (should it ever become one ) to go overlooked.

If the condition were `if str(int) != "<python 3 version>": <python 2 test>" then it's still fragile but a mismatch leads to the test running and (hopefully) failing, thus drawing developer attention.

Alternatively ... I am not against a check against `sys.version_info` for identifying Python versions.  Any reason not to use that (here and possibly throughout)?

Or more alternatively, use of builtin types is not intrinsic to what's being asserted here.  If it simplifies things to use a different type (perhaps a type with the same repr on Python 2 and Python 3), feel free to do that.